run_dir/static/js/data_del_plot.js

```diff
@@ -0,0 +1,357 @@
+            min : 0,
+            title : {
+              text : name
+            }
+        },
+        legend: {
+            title: {
+                text: 'Click to hide:',
+                align: 'center'
+                }
+        },
+        plotOptions : {
+            series : {
+                turboThreshold: 0,
+                point: {
+                  events: {
+                    click: function() {
+                      window.open(this.options.ownURL)
+                    }
+                  }
+                }
+            }
+        },
+        tooltip: {
+            shared: true,
+            useHTML: true,
+            headerFormat: '<span style="color:{point.color}">\u25CF</span><small>{point.key}</small><br />',
+            pointFormatter: function () {
+                if (this.y === 0) return '0 Bytes';
+                const k = 1024;
+                const dm = 2 < 0 ? 0 : 2;
+                const sizes = ['Bytes', 'KB', 'MB', 'GB', 'TB', 'PB', 'EB', 'ZB', 'YB'];
+                const i = Math.floor(Math.log(this.y) / Math.log(k));
+                var fs = parseFloat((this.y / Math.pow(k, i)).toFixed(dm)) + ' ' + sizes[i];
+                return this.series.name + ', ' + this.pm + ': ' + '<b>' + fs + '</b>'
+            },
+        },
+        credits: {
+            enabled: false
+        },
+        xAxis: {
+            type: 'category',
+            labels: {
+                enabled: false,
+            },
+            title : {
+                text : 'Close Date'
+            },
+            categories: []
+        },
+        series: [{
+            name: name,
+            data: []
+        }],
+        exporting: {
+          csv: {
+            itemDelimiter: ';'
+          }
+        }
+    };
+    serie = build_series(window.current_plot_data, key, name, view_type, filter_inst_type);
+    toplot.series = serie[1];
+    toplot.xAxis.categories = serie[0];
+    $("#main_plot").highcharts(toplot);
+    window.current_plot_obj = toplot;
+}
+
+function build_series(data, key, name, view_type, filter_inst_type){
+    var series = [];
+    var view_color = "";
+    var series_name = "";
+    var categories = [];
+    for (d in data){
+        var bioinfo_link="/bioinfo/"+data[d][0];
+        var project_name = data[d][1].project_name;
+        var date_close = data[d][1].close_date;
+        if (data[d][1].sequencing_platform == null){
+            continue;
+        }else if (data[d][1].sequencing_platform.includes('NovaSeq') && filter_inst_type.includes('NovaSeq')){
+            continue;
+        }else if (data[d][1].sequencing_platform.includes('MiSeq') && filter_inst_type.includes('MiSeq')){
+            continue;
+        }else if (data[d][1].sequencing_platform.includes('NextSeq') && filter_inst_type.includes('NextSeq')){
+            continue;
+        }else if (data[d][1].sequencing_platform.includes('HiSeq') && filter_inst_type.includes('HiSeq')){
+            continue;
+        }  
```

Comment:
I think quite a lot of the code below can be refactored to use a pattern like this:

```suggestion
        sequencing_platforms = ['NovaSeq', 'MiSeq', 'NextSeq', 'HiSeq'];
        if (data[d][1].sequencing_platform == null && filter_inst_type.length > 0){
            continue;
        }
        for(var p in sequencing_platforms){
            if (data[d][1].sequencing_platform.includes(p) && filter_inst_type.includes(p)){
                continue;
            }
        }
```

Less repetition so easier to read / maintain.

Note that I'm totally guessing about the `filter_inst_type.length` - that was just my guess to skip data with no seq platform set only if we are filtering for a given platform.

Reply:
Nice!

Reply:
Wait! Just looked at this again and realised that it won't work 😆 The `continue` statement will break the closest `for` loop and so leave the outer `for` loop unaffected. Ok so needs a little tweak, but hopefully the concept is still valid (can use a placeholder `is_skipped` variable or something, whatever).

Reply:
Strange, it didn't work because there are still some "undefined" sequencing platforms even after these should have been caught (?) on the lines before.

This works:

`for (var p in sequencing_platforms){
	    if (data[d][1].sequencing_platform !== undefined && data[d][1].sequencing_platform.includes(p) && filter_inst_type.includes(p)){
	              continue;
	     }
 }`

Reply:
ah ok - maybe better to add that check alongside the `if` statement above which looks for `null` (which is basically the same as `undefined`). But otherwise yeah, looks good 👍🏻

Reply:
I tried, it still gave me an error hmm